Running `yarn lint` fails if `yarn build` not run first

... unless you run `yarn build` first!

That's a bit counterintuitive to me, I'd expect `yarn lint` to work as a "standalone" command.
Is that because of the way the project is configured in packages?

**Repro:**
- Get a fresh clone of the repo
- Run `yarn install`
- Run `yarn lint`
- Error from ESLint:

```
Oops! Something went wrong! :(

ESLint: 6.8.0.

ESLint couldn't find the plugin "@redwoodjs/eslint-plugin-redwood".

(The package "@redwoodjs/eslint-plugin-redwood" was not found when loaded as a Node module from the directory "/Users/olance/Dev/redwood".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install @redwoodjs/eslint-plugin-redwood@latest --save-dev

The plugin "@redwoodjs/eslint-plugin-redwood" was referenced from the config file in "package.json » @redwoodjs/eslint-config".

If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.
```

The plugin was correctly installed though.
After running `yarn build`, then things started working properly.

I quickly talked to some people on ESLint's Gitter room, here's what they said in the end:

> Jordan Harband @ljharb 18:29
> however, it seems the real issue is that @redwoodjs/eslint-config is broken - it should have @redwoodjs/eslint-plugin-redwood listed as a peer dep, and you do have to install it as a dev dep yourself, manually.
> 
> i'd file an issue on that eslint config to make it a proper peer dep
> shared configs must list every plugin they use as a peer dep, it's a limitation of the way shared configs work.
> which is, 10 of them, it looks like. so you have to add all 10 as dev deps to your project
> look at the airbnb config to see how it's done properly

@peterp I think I heard you like configuring stuff? 😄

@olance I believe the issue is here: 

https://github.com/redwoodjs/redwood/blob/a6f6b619206b80e5f6074bcd2f0f37cb610a9124/packages/eslint-config/index.js#L23, and here: https://github.com/redwoodjs/redwood/blob/a6f6b619206b80e5f6074bcd2f0f37cb610a9124/packages/eslint-config/index.js#L46

The library that we're referencing `./pacakges/eslint-plugin-redwood` is not built, and it's not actually required by the monorepo. 
```terminal
@redwoodjs/eslint-config': Cannot find module '/Users/peterp/Personal/redwoodjs/redwood/node_modules/@redwoodjs/eslint-plugin-redwood/dist/index.js'. Please verify that the package.json has a valid "main" entry
```

A simple fix might be to have two configurations:
1. for the redwoodjs framework monorepo
2. for redwoodjs projects  (which includes the `eslint-plugin-redwood`)

A quick fix would be to dynamically require `@redwoodjs/eslint-plugin-redwood` to see if it's available, and to only include it if it is.

--- a/packages/eslint-config/index.js
+++ b/packages/eslint-config/index.js
@@ -10,6 +10,23 @@
 // [^1] https://eslint.org/docs/rules/
 // [^2] https://www.npmjs.com/package/eslint-plugin-react#list-of-supported-rules
 
+let supportRedwoodAutoPageImports = false
+try {
+  // This will throw if the module cannot be resolved, it will not
+  // resolve if the module is not built, in which case we're probably
+  // building the framework and don't need this plugin.
+  require.resolve('@redwoodjs/eslint-plugin-redwood')
+  supportRedwoodAutoPageImports = {
+    files: ['web/src/Routes.js', 'web/src/Routes.ts'],
+    rules: {
+      'no-undef': 'off',
+      '@redwoodjs/redwood/no-unavailable-pages': 'error',
+    },
+  }
+} catch (e) {
+  // noop
+}
+
 module.exports = {
   parser: 'babel-eslint',
   plugins: [
@@ -20,8 +37,8 @@ module.exports = {
     'react',
     'react-hooks',
     'jest-dom',
-    '@redwoodjs/redwood',
-  ],
+    supportRedwoodAutoPageImports && '@redwoodjs/redwood-eslint-plugin',
+  ].filter(Boolean),
   ignorePatterns: ['node_modules', 'dist'],
   extends: [
     'eslint:recommended',
@@ -42,20 +59,14 @@ module.exports = {
         '@typescript-eslint/explicit-function-return-type': 'off',
       },
     },
-    {
-      files: ['web/src/Routes.js', 'web/src/Routes.ts'],
-      rules: {
-        'no-undef': 'off',
-        '@redwoodjs/redwood/no-unavailable-pages': 'error',
-      },
-    },
+    supportRedwoodAutoPageImports,
     {
       files: ['api/src/**'],
       globals: {
         context: 'readonly',
       },
     },
-  ],
+  ].filter(Boolean),
   settings: {
     // This is used to support our `import/order` configuration.
     'import/resolver': {

--- a/packages/eslint-config/index.js
+++ b/packages/eslint-config/index.js
@@ -52,7 +52,6 @@ module.exports = {
     {
       files: ['api/src/**'],
       globals: {
-        db: 'readonly',
         context: 'readonly',
       },
     },

--- a/packages/eslint-config/index.js
+++ b/packages/eslint-config/index.js
@@ -37,7 +37,7 @@ module.exports = {
     'react',
     'react-hooks',
     'jest-dom',
-    supportRedwoodAutoPageImports && '@redwoodjs/redwood-eslint-plugin',
+    supportRedwoodAutoPageImports && '@redwoodjs/redwood', // @redwoodjs/redwood-eslint-plugin
   ].filter(Boolean),
   ignorePatterns: ['node_modules', 'dist'],
   extends: [

